src/realm/object-store/shared_realm.cpp

```diff
@@ -1281,6 +1288,13 @@ void Realm::close()
     m_config = {};
 }
 
+void Realm::close_and_delete_realm()
+{
+    const auto path = m_config.path;
+    close();
+    util::File::remove(path);
```

Comment:
Note, I suspect this might not work for synchronized Realms. We have a dedicated API for deleting a Realm, but using that is difficult since the SyncClient also has a handle to the Realm and https://github.com/realm/realm-core/blob/master/src/realm/object-store/sync/sync_manager.hpp#L157 is too broad.

Reply:
Indeed, I am seeing problems on Windows only. `Unlink` seems to be releases the resources for all the platforms but windows (it is a known problem we have for object store tests too). 
I think we need to release the handle the sync client is holding in this case and delete all. @jbreams  what's the best way for doing this?

Reply:
This will also likely leave a bunch of auxiliary files and directories which we also need to clean up.

Reply:
I am not sure if I follow, if no handles are kept, this API should unlink the whole folder. The only problem I see is for sync realms.

Reply:
Isn't `m_config.path` the path to the realm file itself - e.g. `/documents/foo/default.realm`. Apart from that, there are a bunch of extra files - e.g. `default.lock`, `default.note` as well as some folders - e.g. `default.management` that need to be cleaned up if they were created. You can see a full list in the implementation of `DB::delete_files`.

Reply:
Oh my bad, you are right, we need the folder.

Reply:
We should now have what it is needed in order to delete realm files and stop sync client in case of possible errors during init function.